packages/devtools_app/lib/src/screens/performance/performance_controller.dart

```diff
@@ -611,6 +611,10 @@ class PerformanceController extends DisposableController
     List<TraceEvent> threadNameEvents, {
     bool isInitialUpdate = false,
   }) {
+    // This can happen if there is a race between this method being called and
+    // losing connection to the app.
+    if (serviceManager.connectedApp == null) return;
```

Comment:
Does this squelch something that we would want to reflect?
Perhaps is there an error logging message we could send, or and assertion we could set in case this is ever happening regularly and causing problems?

Reply:
`updateThreadIds` is called from `_pullTraceEventsFromVmTimeline`, which is called using a RateLimiter and polling timer. When we lose connection from the app, we cancel the timer and rate limiter
https://github.com/flutter/devtools/blob/0e3f7e76f9aa1b769267adac524dbf3752393547/packages/devtools_app/lib/src/screens/performance/performance_controller.dart#L241-L244

But the call to `_pullTraceEventsFromVmTimeline` (which calls `updateThreadIds`) could already be in progress. So adding a check that we are still connected here prevents us from calling `serviceManager.connectedApp!` on a null value and unnecessarily continuing on with the rest of the method.